src/wp-admin/js/editor.js

```diff
@@ -116,210 +202,327 @@
 			}
 		}
 
-		// Replace paragraphs with double line breaks
+		/**
+		 * @summary Replaces all paragraphs with double linke breaks.
+		 *
+		 * @since 3.7.10
+		 *
+		 * Replaces all paragraphs with double line breaks. Taking into account
+		 * the elements where the <p> should be preserved.
+		 * Unifies all whitespaces.
+		 * Adds indenting with tabs to li, dt and dd elements.
+		 * Trims whitespaces from beginning and end of the html input.
+		 *
+		 * @param {string} html The content from the editor.
+		 * @return {string} The formatted html string.
+		 */
 		function removep( html ) {
 			var blocklist = 'blockquote|ul|ol|li|dl|dt|dd|table|thead|tbody|tfoot|tr|th|td|h[1-6]|fieldset',
 				blocklist1 = blocklist + '|div|p',
 				blocklist2 = blocklist + '|pre',
 				preserve_linebreaks = false,
-				preserve_br = false,
-				preserve = [];
+				preserve_br = false;
```

Comment:
You are changing the code, I do not think this is a good idea. Just add the docs and nothing else.